package.json

```diff
@@ -39,7 +39,8 @@
     "lint:fix": "eslint . --ext js,json --fix",
     "lint:changed": "{ git ls-files --others --exclude-standard ; git diff-index --name-only --diff-filter=d HEAD ; } | grep --regexp='[.]js$' --regexp='[.]json$' | tr '\\n' '\\0' | xargs -0 eslint",
     "lint:changed:fix": "{ git ls-files --others --exclude-standard ; git diff-index --name-only --diff-filter=d HEAD ; } | grep --regexp='[.]js$' --regexp='[.]json$' | tr '\\n' '\\0' | xargs -0 eslint --fix",
-    "lint:shellcheck": "shellcheck --version && find . -type f -name '*.sh' ! -path './node_modules/*' -print0 | xargs -0 shellcheck",
+    "lint:shellcheck:package": "jq -r '.scripts | keys[] as $key | .[$key]' < package.json > temp.sh && echo '#!/bin/bash' | cat - temp.sh > temp && mv temp temp.sh && shellcheck -e SC2016 temp.sh && rm temp.sh",
```

Comment:
Can we move this into a real script instead of writing it to a temporary one?

Reply:
Would that not sort of default the purpose?

Just in case it isn't clear (I mean, it's kinda confusing just looking at it)... what this is actually doing is taking all the `.scripts` entries and building a temporary file_from those_ that `shellcheck` can then... check... however, maybe there's a cleaner way to do this without creating an actual file on disk? from what I've seen there doesn't appear to be a way for `shellcheck` to lint strings as input which is why I opted for this method.